src/Decoder/SmileDecoder.php

```diff
@@ -183,18 +189,19 @@ private function decodeInt(): int
         return $result;
     }
 
-    private function decodeBigInt(): int
+    private function decodeBigInt(): float
     {
         $int = $this->decodeInt();
 
-        $bytesAmount = round((($int * 8) / 7));
+        $bytesAmount = (int) ceil(((float) $int * 8) / 7);
 
         $binaryString = '';
 
-        foreach (range(1, $bytesAmount) as $index) {
-            $binaryString .= sprintf('%07b', $this->context->getSpecificByte($this->context->getIndex() + $index));
+        for ($i = 0; $i < $bytesAmount; ++$i) {
```

Comment:
This is strange, you use to have `1` as `$index` during the first loop. Now you have 0.
same for bytesAmount vs bytesAmount - 1

Reply:
This is because before we stopped the loop when we met `byteAmount`, now we stop the loop when we meet `byteAmount - 1`